Permafail E2E: fix Google Authentication spec.

worldomonation opened the pull request:
Fixes https://github.com/Automattic/wp-calypso/issues/75674.


## Proposed Changes

This PR fixes the permanently failing Google Authentications spec within the Authentications suite of specs.

Key changes:
- update POMs to use accessible locators.
- codify expected URL in the test steps.
- handle the double 2FA challenge that Google is sending our test account. I have not seen this in actual user accounts before and I have structured the E2E spec to handle the scenario where Google returns to a single 2FA challenge.

## Testing Instructions

Ensure the following build configurations are passing:
  - [x] Authentication E2E

Authentication E2E suite (limited to the Google Authenticate scenario, to save on SMS costs) passed:
![image](https://github.com/Automattic/wp-calypso/assets/6549265/66a20721-f21d-4600-868f-8fc484ec7b3e)


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
packages/calypso-e2e/src/lib/pages/login-page.ts
- use accessible selector to locate the Login with Google button.

packages/calypso-e2e/src/lib/pages/external/google-login-page.ts
- update methods to use accessibble locators.

worldomonation committed:
test/e2e/specs/authentication/authentication__google.ts
- update step names.
- handle double 2FA challenge that Google has been throwing at us, where two 2FA challenges are shown for TOTP.
- codify expected URL paths.

karenroldan commented on test/e2e/specs/authentication/authentication__google.ts:
```diff
@@ -8,67 +8,105 @@
+	// The result is that the user is required to enter two sets of
+	// TOTP codes, and because the TOTP codes regenerate every 30s
+	// this means the spec has to wait until the TOTP code updates.
+	// This behavior was confirmed manually by @worldomonation as well.
```
**praise**: thank you for the detailed context on why this is added.

karenroldan reviewed (approved):
LGTM 🚀

worldomonation committed:
test/e2e/specs/authentication/authentication__google.ts
- add a boatload of comments explaining the reasoning behind each line.
- fix logic issue with waiting on `page.on` - this should have been a `googlePopupPage.on`.
